src/frontends/tensorflow/src/helper_ops/block_lstm.hpp

```diff
@@ -0,0 +1,143 @@
+              const Output<Node>& wco,
+              const Output<Node>& b,
+              float forget_bias,
+              float cell_clip,
+              bool use_peephole,
+              const std::shared_ptr<DecoderBase>& decoder = nullptr)
+        : InternalOperation(decoder, OutputVector{seq_len_max, x, cs_prev, h_prev, w, wci, wcf, wco, b}, 7),
+          m_forget_bias(forget_bias),
+          m_cell_clip(cell_clip),
+          m_use_peephole(use_peephole),
+          m_hidden_size(ov::Dimension::dynamic()) {
+        validate_and_infer_types();
+    }
+
+    void validate_and_infer_types() override {
+        // BlockLSTM description
+        //
+        // Inputs:
+        // 0) seq_len_max	A Tensor of type int64. Maximum time length actually used by this input. Outputs are
+        // padded with zeros beyond this length.
+        // 1) x A Tensor. Must be one of the following types: half, float32. The sequence input to the LSTM, shape
+        // (timelen, batch_size, num_inputs).
+        // 2) cs_prev A Tensor. Must have the same type as x. Value of the initial cell state.
+        // 3) h_prev A Tensor. Must have the same type as x. Initial output of cell (to be used for peephole).
+        // 4) w	A Tensor. Must have the same type as x. The weight matrix.
+        // 5) wci A Tensor. Must have the same type as x. The weight matrix for input gate peephole connection.
+        // 6) wcf A Tensor. Must have the same type as x. The weight matrix for forget gate peephole connection.
+        // 7) wco A Tensor. Must have the same type as x. The weight matrix for output gate peephole connection.
+        // 8) b	A Tensor. Must have the same type as x. The bias vector.
+        //
+        // Attributes:
+        // forget_bias	An optional float. Defaults to 1. The forget gate bias.
+        // cell_clip	An optional float. Defaults to 3. Value to clip the 'cs' value to.
+        // use_peephole	An optional bool. Defaults to False. Whether to use peephole weights.
+        //
+        // Outputs:
+        // 0) i A Tensor. Has the same type as x.
+        // 1) cs A Tensor. Has the same type as x.
+        // 2) f A Tensor. Has the same type as x.
+        // 3) o A Tensor. Has the same type as x
+        // 4) ci A Tensor. Has the same type as x.
+        // 5) co A Tensor. Has the same type as x.
+        // 6) h A Tensor. Has the same type as x.
+
+        // extract time_len, batch_size
+        auto time_len = ov::Dimension::dynamic();
+        auto batch_size = ov::Dimension::dynamic();
+        auto x_shape = get_input_partial_shape(1);
+        auto x_type = get_input_element_type(1);
+        auto x_rank = x_shape.rank();
+        if (x_rank.is_static()) {
+            FRONT_END_OP_CONVERSION_CHECK(
+                x_rank.get_length() == 3,
+                "Internal error in OpenVINO TensorFlow Frontend: input data for BlockLSTM must be of rank equal to 3.");
+            time_len = x_shape[0].get_length();
```

Comment:
Please check that this dim is static
![image](https://user-images.githubusercontent.com/21181928/191911691-8fe3feb5-919e-44d9-bca0-3be442a1767f.png)

Reply:
Great catch! Done